### PR TITLE
Task 1209

### PR DESCRIPTION
**Pull request by krozzus**

### CODE REVIEW

File: rest/taskana-rest-spring/src/main/java/pro/taskana/rest/common/AccessIdController.java [hunk 1]
```diff
@@ -26,8 +26,13 @@
...
 
   private static final Logger LOGGER = LoggerFactory.getLogger(AccessIdController.class);
+  private static final String EXIT_FROM_VALIDATE_ACCESS_IDS
+      = "Exit from validateAccessIds(), returning {}";
+  private static final String EXIT_FROM_GET_GROUP_BY_ACCESS_ID
+      = "Exit from getGroupsByAccessId(), returning {}";
   private static LdapCache ldapCache;
-  @Autowired LdapClient ldapClient;
+  @Autowired
+  LdapClient ldapClient;
```
Comment (mustaphazorgati) [inline rest/taskana-rest-spring/src/main/java/pro/taskana/rest/common/AccessIdController.java:35]: Please use an autowired constructor ;)

File: rest/taskana-rest-spring/src/main/java/pro/taskana/rest/common/TaskanaErrorData.java [hunk 2]
```diff
@@ -25,7 +25,7 @@
     this.exception = ex.getClass().getName();
     this.message = ex.getMessage();
     this.path = req.getDescription(false);
-    if (this.path != null && this.path.startsWith("uri=")) {
+    if (this.path.startsWith("uri=")) {
```
Comment (mustaphazorgati) [inline rest/taskana-rest-spring/src/main/java/pro/taskana/rest/common/TaskanaErrorData.java:28]: why did you remove the null check?

File: rest/taskana-rest-spring/src/main/java/pro/taskana/rest/common/TaskanaRestExceptionHandler.java [hunk 3]
```diff
@@ -37,7 +37,8 @@
 @ControllerAdvice
 public class TaskanaRestExceptionHandler extends ResponseEntityExceptionHandler {
 
-  private static final Logger LOGGER = LoggerFactory.getLogger(TaskanaRestExceptionHandler.class);
+  private static final Logger REST_EXCEPTION_HANDLER_LOGGER = LoggerFactory.getLogger(
```
Comment (mustaphazorgati) [inline rest/taskana-rest-spring/src/main/java/pro/taskana/rest/common/TaskanaRestExceptionHandler.java:40]: What's wrong with the name LOGGER?

File: rest/taskana-rest-spring/src/main/java/pro/taskana/rest/common/models/PagedResources.java [hunk 4]
```diff
@@ -61,8 +61,8 @@ public PagedResources(Collection<T> content, PageMetadata metadata, Iterable<Lin
   @JsonProperty("page")
   public PageMetadata getMetadata() {
     if (Objects.isNull(metadata)) {
-      Collection<T> content = getContent();
-      return new PageMetadata(content.size(), 0, content.size());
+      Collection<T> contentCollection = getContent();
+      return new PageMetadata(contentCollection.size(), 0, contentCollection.size());
     }
     return metadata;
   }
```
Comment (mustaphazorgati) [inline rest/taskana-rest-spring/src/main/java/pro/taskana/rest/common/models/PagedResources.java:68]: This is an artifact for taskana-history, right?

File: rest/taskana-rest-spring/src/main/java/pro/taskana/rest/monitor/MonitorController.java [hunk 5]
```diff
@@ -30,9 +31,9 @@
 
   private static final Logger LOGGER = LoggerFactory.getLogger(MonitorController.class);
 
-  private MonitorService monitorService;
+  private final MonitorService monitorService;
 
-  private ReportRepresentationModelAssembler reportRepresentationModelAssembler;
+  private final ReportRepresentationModelAssembler reportRepresentationModelAssembler;
 
   MonitorController(
```
Comment (mustaphazorgati) [inline rest/taskana-rest-spring/src/main/java/pro/taskana/rest/monitor/MonitorController.java:38]: Can you please add the `@Autowired` annotation here?

File: rest/taskana-rest-spring/src/main/java/pro/taskana/rest/workbasket/WorkbasketDefinitionController.java [hunk 6]
```diff
@@ -159,10 +160,8 @@
       boolean noWrongAuth =
           definition.getAuthorizations().stream()
               .noneMatch(
-                  access -> {
-                    return (!access.getWorkbasketId().equals(importedWb.getId()))
-                        || (!access.getWorkbasketKey().equals(importedWb.getKey()));
-                  });
+                  access -> (!access.getWorkbasketId().equals(importedWb.getId()))
+                                || (!access.getWorkbasketKey().equals(importedWb.getKey())));
       if (!noWrongAuth) {
```
Comment (mustaphazorgati) [inline rest/taskana-rest-spring/src/main/java/pro/taskana/rest/workbasket/WorkbasketDefinitionController.java:165]: Can you please further simplify this:
an inverse as a variable name is not clean.
- rename the variable to `authenticated`
- replace noneMatch with anymatch
- invert predicate

Comment (mustaphazorgati) [reply rest/taskana-rest-spring/src/main/java/pro/taskana/rest/workbasket/WorkbasketDefinitionController.java:165]: The error message below has to be adjusted too then :)

File: rest/taskana-rest-spring/src/main/java/pro/taskana/rest/workbasket/models/WorkbasketRepresentationModel.java [hunk 7]
```diff
@@ -13,7 +13,7 @@
   private String modified; // ISO-8601
 
   @SuppressWarnings("unused") //Mandatory for Jackson
-  protected WorkbasketRepresentationModel() {
+  public WorkbasketRepresentationModel() {
```
Comment (mustaphazorgati) [inline rest/taskana-rest-spring/src/main/java/pro/taskana/rest/workbasket/models/WorkbasketRepresentationModel.java:16]: why did you make this constructor public?

File: rest/taskana-rest-spring/src/test/java/pro/taskana/rest/monitor/models/ReportRepresentationModelTest.java [hunk 8]
```diff
@@ -22,12 +22,16 @@
...
-/** Test for {@link ReportRepresentationModelAssembler}. */
+/**
+ * Test for {@link ReportRepresentationModelAssembler}.
+ */
 @TaskanaSpringBootTest
 class ReportRepresentationModelTest {
 
-  @Autowired private ReportRepresentationModelAssembler reportRepresentationModelAssembler;
+  @Autowired
+  private ReportRepresentationModelAssembler reportRepresentationModelAssembler;
```
Comment (mustaphazorgati) [inline rest/taskana-rest-spring/src/test/java/pro/taskana/rest/monitor/models/ReportRepresentationModelTest.java:34]: Please use an autowired constructor

File: rest/taskana-rest-spring/src/test/java/pro/taskana/rest/workbasket/WorkbasketDefinitionControllerIntTest.java [hunk 9]
```diff
@@ -1,289 +1,290 @@
...
+
+  private String workbasketToString(
+      WorkbasketDefinitionRepresentationModel workbasketDefinitionRepresentationModel) {
+    try {
+      return objMapper.writeValueAsString(workbasketDefinitionRepresentationModel);
+    } catch (JsonProcessingException e) {
+      return "";
+    }
+  }
+}
```
Comment (mustaphazorgati) [inline rest/taskana-rest-spring/src/test/java/pro/taskana/rest/workbasket/WorkbasketDefinitionControllerIntTest.java:290]: What changed here? Why is the diff so huge?

File: rest/taskana-rest-spring/src/main/java/pro/taskana/rest/common/TaskanaErrorData.java (hunk 2 above)
Comment (krozzus) [reply rest/taskana-rest-spring/src/main/java/pro/taskana/rest/common/TaskanaErrorData.java:28]: According to the IntelliJ and Sonarcloud the method getDescription never returns null and != null is thereby always true.

File: rest/taskana-rest-spring/src/main/java/pro/taskana/rest/common/TaskanaRestExceptionHandler.java (hunk 3 above)
Comment (krozzus) [reply rest/taskana-rest-spring/src/main/java/pro/taskana/rest/common/TaskanaRestExceptionHandler.java:40]: This is a suggestion by sonarcloud to prevent a mixup with the field of the superclass.

File: rest/taskana-rest-spring/src/main/java/pro/taskana/rest/common/models/PagedResources.java (hunk 4 above)
Comment (krozzus) [reply rest/taskana-rest-spring/src/main/java/pro/taskana/rest/common/models/PagedResources.java:68]: This is a suggestion by sonarcloud to make sure the field with the same name isn't hidden.
